kew.js

```diff
@@ -299,6 +299,38 @@ Promise.prototype.failBound = function (onFail, scope, var_args) {
 }
 
 /**
+ * Spread a promises outputs to the functions arguments.
+ * @param {?function(this:void, ...): RESULT|undefined} onSuccess
+ * @return {!Promise.<RESULT>} returns a new promise with the output of the onSuccess or
+ *     onFail handler
+ * @template RESULT
+ */
+Promise.prototype.spread = function (onSuccess) {
+  return this
+  .then(allInternal)
```

Comment:
Can you put this on the same line as `this`? I find this construction very hard to parse. Ditto below